packages/http-server/src/http-server.ts

```diff
@@ -109,6 +119,26 @@ export class HttpServer {
     } else {
       this.server = http.createServer(this.requestListener);
     }
+
+    // Apply server properties
+    const {keepAliveTimeout, headersTimeout, maxConnections, maxHeadersCount,
+      timeout} = this.serverOptions;
+    if (keepAliveTimeout) {
+      this.server.keepAliveTimeout = keepAliveTimeout;
+    }
+    if (headersTimeout) {
+      this.server.headersTimeout = headersTimeout;
+    }
+    if (maxConnections) {
+      this.server.maxConnections = maxConnections;
+    }
+    if (maxHeadersCount) {
+      this.server.maxHeadersCount = maxHeadersCount;
+    }
+    if (timeout) {
+      this.server.timeout = timeout;
+    }
```

Comment:
Can you please double-check that all of these properties are supported by Node.js 10.x (the oldest version we support)?

It would be better to have an integration test that sets the option and then observes that the server is honoring it (behaving differently when compared to default).

Reply:
All of those properties exist on 10.x:
- https://nodejs.org/dist/latest-v10.x/docs/api/http.html
- https://nodejs.org/dist/latest-v10.x/docs/api/net.html

Reply:
I propose to use the following singature:

```ts
/**
 * The following are for configuring properties which are directly set on
 * https://nodejs.org/api/http.html#http_class_http_server and
 * https://nodejs.org/api/net.html#net_class_net_server
 */
export type HttpServerProperties = Partial<
  Pick<
    Server,
    | 'keepAliveTimeout'
    | 'headersTimeout'
    | 'maxConnections'
    | 'maxHeadersCount'
    | 'timeout'
  >
>;

/**
 * Base options that are common to http and https servers
 */
export interface BaseHttpOptions extends ListenOptions, HttpServerProperties {
  /**
   * The `gracePeriodForClose` property controls how to stop the server
   * gracefully. Its value is the number of milliseconds to wait before
   * in-flight requests finish when the server is being stopped. With this
   * setting, we also reject new requests from existing keep-alive connections
   * in addition to stopping accepting new connections.
   *
   * Defaults to Infinity (don't force-close). If you want to immediately
   * destroy all sockets set its value to `0`.
   *
   * See {@link https://www.npmjs.com/package/stoppable | stoppable}
   */
  gracePeriodForClose?: number;
}
```